### PR TITLE
Quickbooks kick off

### PR DESCRIPTION
**Pull request by janprochazkacz**

### CODE REVIEW

**Comment by UBNT-Petr** [comment]
needs some Readme file to guide users:
This section is used to integrate UCRM with QuickBooks (oneway data migration from UCRM to QuickBooks)
1. First, configure your account settings in...
2. Use payments.php to import selected payments from UCRM to QB
3. Use ....

File: config.dist/quickbooks.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+// ClientID
+const QUICKBOOKS_OAUTH_CLIENTID = 'PUT YOUR CLIENT ID HERE';
+
+// Client Secret
+const QUICKBOOKS_OAUTH_CLIENTSECRET = 'PUT YOUR CLIENT SECRET HERE';
+
+// OAuth2 Redirect URI
+const QUICKBOOKS_OAUTH_REDIRECT_URI = 'http://b3286fda.ngrok.io';
```
Comment (keksa) [inline config.dist/quickbooks.php]: what should we put here?

File: config.dist/quickbooks.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+// BaseURL "Development" or "Production"
+const QUICKBOOKS_BASE_URL = 'Production';
+
+
+// Values returned by QB after confirming connection on URL generated with OAuth2UrlGenerator.php
+const QUICKBOOKS_OAUTH_CODE = 'L011514360516YCAj1da2FEllWjYUBkVbnyH7ARuNWmJEp9eqN';
```
Comment (keksa) [inline config.dist/quickbooks.php:22]: where do I get these values?

File: scripts/quickbooksonline_import/invoices.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    'accessTokenKey' => QUICKBOOKS_OAUTH_ACCESS_TOKEN,
+    'refreshTokenKey' => QUICKBOOKS_OAUTH_REFRESH_TOKEN,
+    'baseUrl' => QUICKBOOKS_BASE_URL,
+    'QBORealmID' => QUICKBOOKS_OAUTH_REALMID,
+));
+
+
+function getQBClient($dataService, $ucrmClientId) {
+    $customers = $dataService->Query(
+        sprintf('SELECT * FROM Customer WHERE DisplayName LIKE \'%%UCRMID-%d%%\'', $ucrmClientId)
```
Comment (keksa) [inline scripts/quickbooksonline_import/invoices.php]: maybe limit 1?

File: scripts/quickbooksonline_import/invoices.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+function getInvoicesFromUCRM() {
+    return ucrmApiQuery('invoices');
+}
+
+function createQBLineFromItem($dataService, $item) {
+    $item = Item::create([
+        'Name' => sprintf('%s (UCRMID-%s)', $item['label'], $item['id']) ,
+        'Type' => 'Service',
+        'IncomeAccountRef' => [
+            'value' => 30
```
Comment (keksa) [inline scripts/quickbooksonline_import/invoices.php]: why 30?

**Review by keksa — CHANGES_REQUESTED** [review]

**Pull request closed without merging**